server/src/main/java/org/opensearch/index/codec/CodecService.java

```diff
@@ -104,4 +110,9 @@ public Codec codec(String name) {
     public String[] availableCodecs() {
         return codecs.keySet().toArray(new String[0]);
     }
+
+    public static boolean isZStandardCodec(String codec) {
+        return codec.equals(ZSTD_CODEC) || codec.equals(ZSTD_NO_DICT_CODEC);
+    }
```

Comment:
Can this be done in a generic way to check if the codec supports compression level or not. Even, more generically if setting X is supported in codec Y?

Reply:
+1

Reply:
@backslasht added a generic way to validate settings